src/actions/notes.js

```diff
@@ -13,49 +14,63 @@ export const CREATE_NOTE_START = 'CREATE_NOTE_START';
 export const CREATE_NOTE_SUCCESS = 'CREATE_NOTE_SUCCESS';
 export const CREATE_NOTE_FAILURE = 'CREATE_NOTE_FAILURE';
 
-const postUpdateNote = (bookmarkId, noteId, noteFields) => ({
-  [CALL_API]: {
-    types: [
-        { type: UPDATE_NOTE_START, meta: { bookmarkId, noteId } },
-        { type: UPDATE_NOTE_SUCCESS, meta: { bookmarkId, noteId } },
-        { type: UPDATE_NOTE_FAILURE, meta: { bookmarkId, noteId } },
-    ],
-    method: 'POST',
-    body: JSON.stringify(noteFields),
-    endpoint: `${API_VERSION}/notes/${noteId}`,
-  },
-});
-
-const postDeleteNote = (bookmarkId, noteId) => ({
-  [CALL_API]: {
-    types: [
-        { type: DELETE_NOTE_START, meta: { bookmarkId, noteId } },
-        { type: DELETE_NOTE_SUCCESS, meta: { bookmarkId, noteId } },
-        { type: DELETE_NOTE_FAILURE, meta: { bookmarkId, noteId } },
-    ],
-    method: 'DELETE',
-    endpoint: `${API_VERSION}/notes/${noteId}`,
-  },
-});
-
-const postCreateNote = (bookmarkId, noteFields) => ({
-  [CALL_API]: {
-    types: [
-        { type: CREATE_NOTE_START, meta: { bookmarkId } },
-        { type: CREATE_NOTE_SUCCESS, meta: { bookmarkId } },
-        { type: CREATE_NOTE_FAILURE, meta: { bookmarkId } },
-    ],
-    method: 'POST',
-    body: JSON.stringify(noteFields),
-    endpoint: `${API_VERSION}/bookmarks/${bookmarkId}/notes`,
-  },
-});
-
-export const updateNote = (bookmarkId, noteId, noteFields) => (dispatch) =>
-    dispatch(postUpdateNote(bookmarkId, noteId, noteFields));
+
+const postCreateNote = (bookmarkId, { text, tmpId }) => {
```

Comment:
Can we generate `tmpId` right here instead of passing it in?

Reply:
Also, it is better to use `noteFields` without destructuring. (More generic and won't be changed after adding new fields to note.)

Reply:
I prefer to have explicit parameters.

Reply:
> Can we generate tmpId right here instead of passing it in?
Yes.

Reply:
> I prefer to have explicit parameters.

Imagine adding `tags` field to notes. How many places must you change? Adding `parent` field? Adding `noteType`? (arbitrary example)

Reply:
By explicitness you do not mean named arguments, do you?

Reply:
All changes will be in one file, that's fine to me.

I do not want to look for action method usage in order to understand what is coming to reducer. 

If we use flowjs and described the params as a type I would be in favor of the suggestion.

Reply:
> By explicitness you do not mean named arguments, do you?

By explicitly I mean, say `{ text, tmpId }` instead of `noteFields`.

Reply:
>Imagine adding tags field to notes. How many places must you change? Adding parent field? Adding reviewedDate?

OK, if we write reducers in more generic way, then it makes more sense to not destruct params in action.